sdk/helper/clientcountutil/clientcountutil.go

```diff
@@ -0,0 +1,385 @@
+// Package clientcountutil provides a library to generate activity log data for
+// testing.
+package clientcountutil
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/sdk/helper/clientcountutil/generation"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+// ActivityLogDataGenerator holds an ActivityLogMockInput. Users can create the
+// generator with NewActivityLogData(), add content to the generator using
+// the fluent API methods, and generate and write the JSON representation of the
+// input to the Vault API.
+type ActivityLogDataGenerator struct {
+	data            *generation.ActivityLogMockInput
+	addingToMonth   *generation.Data
+	addingToSegment *generation.Segment
+	client          *api.Client
+}
+
+// NewActivityLogData creates a new instance of an activity log data generator
+// The type returned by this function cannot be called concurrently
+func NewActivityLogData(client *api.Client) *ActivityLogDataGenerator {
+	return &ActivityLogDataGenerator{
+		client: client,
+		data:   new(generation.ActivityLogMockInput),
+	}
+}
+
+// NewCurrentMonthData opens a new month of data for the current month. All
+// clients will continue to be added to this month until a new month is created
+// with NewPreviousMonthData.
+func (d *ActivityLogDataGenerator) NewCurrentMonthData() *ActivityLogDataGenerator {
+	d.newMonth(&generation.Data{Month: &generation.Data_CurrentMonth{CurrentMonth: true}})
+	return d
```

Comment:
Totally optional: what do you think about combining these two lines in the same style as `NewPreviousMonthData()` below?